### PR TITLE
Add support for "other" interpolated responses

### PR DESCRIPTION
**Pull request by robcole**
Adds support for interpolating answers in Q&A format of
"other" responses in questions.

### CODE REVIEW

File: lib/typed_form/form_data/answers.rb [hunk 1]
```diff
@@ -82,7 +82,7 @@ def find_answer_by_field_id(id)
         end
 
         answers_found = fields.map { |f| find_answer_by_id(f.id) }.compact
-        return find_answer_by_id(fields.first.id) if answers_found.size == 1
+        return answers_found.first if answers_found.size == 1
```
Comment (timimsms) [inline lib/typed_form/form_data/answers.rb]: 💥

Comment (timimsms) [reply lib/typed_form/form_data/answers.rb]:
```ruby
return answers_found.first unless answers_found.empty?
```

May be slightly more readable.

**Comment by robcole** [comment]
No, because we're actually controlling for answers > 1 and and answers < 1,
not just answers < 1. The main concern is that it's theoretically possible
(who knows with Typeform data) to find multiple answers on a given key,
since the keys aren't guaranteed unique. I've yet to see anything other
than answers == 1 but given the way their data has historically behaved,
leaving the guard clause as-is for now.

**Comment by timimsms** [comment]
Ah - makes sense. I realized this wouldn't behave the same way / as needed given how we're using the guard clause after making this comment. Sorry about that!

**Pull request merged**